### ISSUE TITLE
[Gen2 CF-HPG v1.3 Run] Execute preregistered multiscale-readout seed42 validation experiment

### ISSUE THREAD

**Issue by Irthn1311** [Irthn1311/FER2013_Graph#53]
## Objective

Execute exactly one preregistered **CF-HPG v1.3 Multiscale Readout seed-42 FER2013 train/validation run** after independent implementation review and merge of PR #52.

Primary hypothesis:

> Accepted v1.2 remained underfit and its classifier consumed only the final coarse graph summary. If information learned at the 144-node fine graph stage is being lost by fixed 2x2 averaging/coarse reasoning, concatenating parameter-free fine mean/max with the unchanged coarse mean/max should improve clean-train fit and validation performance without changing tokenizer, graph construction, message passing, hierarchy, or training protocol.

This issue authorizes **one scientific training/validation attempt only**. It does not authorize test access, hyperparameter search, automatic retry, resume, architecture changes, or v1.4.

## Exact authorized source

Run only from exact merged `main` commit:

`8468dfe40ae2e6229c835b3356bb61eac1f22c77`

Reviewed PR #52 head:

`a35e641b4a5ddf4f7c3aafa4379bbd7afffd31bf`

Implementation base / accepted v1.2 source:

`032c38d49d189000b364d24af7e248eba200acf7`

Registered v1.3 model source SHA-256:

`2001fa00a2c21400817dfcd0857e1f3dfe1e0f7979039eee2efa7b7123160495`

Registered model inventory:

- parameters: `445799`
- trainable variables: `66`
- Keras variables: `76`
- input `48x48x1`
- patch `4x4`, 144 fine nodes
- tokenizer `Dense16->96 -> GELU -> Dense96->96`
- positional projection `2->96`, additive after raw-local refinement
- exactly 2 fine Max-Relative blocks, D=96
- hybrid spatial-8 UNION cosine-top4 fine graph, built once/reused twice
- fine readout after second fine block: `mean96 + max96 = 192`
- exact parameter-free `2x2` arithmetic-mean hierarchy to 36 coarse nodes
- coarse projection `96->128`
- exactly 2 coarse Max-Relative blocks, D=128
- same hybrid coarse graph, built once/reused twice
- coarse readout: `mean128 + max128 = 256`
- fused readout order: `[fine_mean, fine_max, coarse_mean, coarse_max] = 448`
- `LayerNorm(448) -> Dense(448->128) -> GELU -> Dropout(0.20) -> Dense(128->7)`

No source/model/training-config change is authorized here.

## Locked v1.2 comparison reference

Accepted Issue #50 seed42:

- clean-train accuracy `0.614894284022432`
- clean-train macro-F1 `0.5614369765915708`
- validation accuracy `0.5806631373641683`
- validation macro-F1 `0.5238975323290902`

Exact deltas:

- `delta_val_accuracy_pp = 100*(v1.3_val_acc - 0.5806631373641683)`
- `delta_val_macro_pp = 100*(v1.3_val_macro - 0.5238975323290902)`
- `delta_clean_train_accuracy_pp = 100*(v1.3_train_acc - 0.614894284022432)`
- `delta_clean_train_macro_pp = 100*(v1.3_train_macro - 0.5614369765915708)`

## Dataset boundary

Dataset signature `fer2013_train28709_val3589_test3589`.

Allowed only:
- train exactly `28709`
- validation exactly `3589`

Absolutely forbidden:
- open/read/hash/load `test.csv`
- test labels/predictions/metrics/confusion matrix
- test smoke/debug/model selection
- renaming/copying test to bypass guards

Any fail-closed path guard failure => STOP.

## Exact training configuration

Frozen from accepted v1.2:

- seed `42`
- AdamW
- LR `3e-4`
- weight decay `5e-4`
- global clipnorm `1.0`
- batch `64`
- max epochs `100`
- linear warmup `5`
- cosine decay to `1e-6`
- CE from logits with label smoothing `0.05`
- earliest strict max `val_accuracy` checkpoint
- early stopping on `val_loss`, patience `15`, min_delta `0`
- same train-only augmentation as v1.2
- validation and clean-train augmentation OFF

No MixUp/CutMix/SAM/TTA/class weights/oversampling/alternate optimizer/alternate seed.

## Exact authorized command

From exact checkout `8468dfe40ae2e6229c835b3356bb61eac1f22c77`:

```bash
python -m research.candidates.tf_cf_hpg_v1_3_multiscale_readout.train_validation_only \
  --train-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/train.csv \
  --val-csv /kaggle/input/datasets/doduyquynii/fer13-split/fer13-split/val.csv \
  --output-root /kaggle/working/tf_cf_hpg_v1_3_multiscale_readout_seed42
```

Do not add/change flags.

## One-attempt rule

Checkout/auth/dataset/runtime preflight stops before the training command starts do not consume the attempt.

Once the training command starts, the one authorized scientific attempt is consumed. After completion, early stop, crash, OOM, CUDA failure or Kaggle censor: STOP. Do not rerun, resume or modify source/config. Any second attempt requires a new written amendment justified by technical invalidity, never by performance.

## Runtime preflight

Before launch record:
- exact HEAD and clean tracked worktree
- model SHA-256
- params / trainable-variable count / Keras-variable count
- Python / TensorFlow / Keras
- GPU model/count
- CUDA/cuDNN if available
- mixed-precision policy
- external XLA/JIT state if available

Do not externally enable mixed precision or XLA/JIT. TensorFlow-internal XLA compilation may be recorded as a runtime note.

## Registered endpoints

Primary: validation accuracy at the selected earliest-strict-max-val-accuracy checkpoint.

Secondary:
- validation macro-F1/loss
- clean-train accuracy/macro-F1/loss
- accuracy/macro clean gaps
- selected epoch / epochs completed
- exact four deltas vs v1.2
- exact model identity

`acc_gap_pp = 100*(clean_train_accuracy - validation_accuracy)`

`macro_gap_pp = 100*(clean_train_macro_f1 - validation_macro_f1)`

## Frozen decision precedence

1. `CF_HPG_V1_3_STRETCH_PASS`
2. `CF_HPG_V1_3_PASS`
3. `READOUT_OVERFIT_SHIFT`
4. `READOUT_STRONG_SIGNAL`
5. `READOUT_PARTIAL_SIGNAL`
6. `READOUT_UNDERFIT_REMAINS`
7. `READOUT_INCONCLUSIVE`

### STRETCH_PASS
All: val acc >= `0.7000`; val macro >= `0.6700`; both gaps <= `8 pp`.

### PASS
All: val acc >= `0.6500`; val macro >= `0.6200`; both gaps <= `8 pp`.

### READOUT_OVERFIT_SHIFT
clean-train accuracy improvement vs v1.2 >= `5.0 pp` AND either clean gap > `10 pp`.

### READOUT_STRONG_SIGNAL
val accuracy improvement vs v1.2 >= `5.0 pp` AND clean-train accuracy improvement vs v1.2 >= `5.0 pp`, unless a higher-priority label applies.

### READOUT_PARTIAL_SIGNAL
val accuracy improvement vs v1.2 >= `3.0 pp` and < `5.0 pp`, unless higher-priority label applies.

### READOUT_UNDERFIT_REMAINS
clean-train accuracy < `0.7000` AND validation accuracy < `0.6400`, unless higher-priority label applies.

Else `READOUT_INCONCLUSIVE`.

These are single-seed practical/diagnostic labels, not statistical superiority/equivalence claims.

## Required evidence

Preserve fresh output root and full available training log. Require at minimum:
- `resolved_config.json`
- `model_summary.txt`
- `checkpoints/best_val_accuracy.keras`
- `checkpoints/best_val_accuracy.metadata.json`
- `validation_only_result.json`
- exact command / return code / runtime identity
- explicit `test_access=false`

After termination package output root once, compute ZIP SHA-256, bytes, member count, outer ZIP integrity and embedded `.keras` integrity. Do not repack after final hash.

If no valid final result exists, preserve partial artifacts as technical evidence only and do not infer a scientific label.

## Interpretation boundary

No test evaluation and no v1.4 before independent evidence review. PASS/STRETCH would justify confirmatory seeds and graph ablations; other outcomes determine the next single hypothesis only after review.

**Comment by Irthn1311** [comment]
## Independent research-lead evidence review — ACCEPTED

I independently inspected the uploaded Issue #53 evidence rather than relying on the execution summary.

### Provenance / integrity
- exact source checkout: `8468dfe40ae2e6229c835b3356bb61eac1f22c77`
- model SHA-256: `2001fa00a2c21400817dfcd0857e1f3dfe1e0f7979039eee2efa7b7123160495`
- model identity: `445799 / 66 / 76`
- train/val counts: `28709 / 3589`
- process return code `0`
- wrapper `COMPLETE`, `scientific_result_valid=true`, `test_access=false`
- evidence ZIP independently recomputed as `5131194` bytes with SHA-256 `1e93a3428b94fc260a127874cb5a150e03d216f9a0b8408a455f474a6f526945`
- outer ZIP CRC PASS; exactly 9 members
- embedded checkpoint `.keras` SHA-256 independently recomputed as `af7cbae14ab769ff7a30ce316fb607314837ae1bc976ab8d4b86dba2ee98e451`; nested ZIP integrity PASS with 3 members
- archive contains no train/val/test CSV

### Accepted scientific result
Selected checkpoint is zero-based epoch `66` (human epoch 67), earliest strict max validation accuracy.

Clean train:
- accuracy `0.5996725765439409`
- macro-F1 `0.5401031020689397`
- loss `1.157474390810745`

Validation:
- accuracy `0.5762050710504318`
- macro-F1 `0.5037193741523447`
- loss `1.2250360376026412`

Recomputed clean gaps:
- accuracy `2.3467505493509067 pp`
- macro-F1 `3.6383727916595054 pp`

Recomputed deltas vs accepted v1.2:
- val accuracy `-0.4458066313736464 pp`
- val macro-F1 `-2.017815817674551 pp`
- clean-train accuracy `-1.5221707478491053 pp`
- clean-train macro-F1 `-2.1333874522631024 pp`

Registered decision is correctly `READOUT_UNDERFIT_REMAINS`.

### Training-lifecycle review
The training log contains 91 epoch summaries. Rounded maximum `val_accuracy=0.5762` occurs at epoch 67. Minimum rounded `val_loss=1.2100` occurs at epoch 76; epochs 77–91 provide the full 15-epoch non-improvement patience window, so early stopping at 91 is consistent. There is no evidence that the selected validation-accuracy checkpoint was missed because of premature stopping.

TensorFlow's single internal XLA-cluster compilation is accepted as the same runtime note class seen in prior runs; no external XLA/JIT change was made.

### Interpretation
This intervention does **not** support coarse-only readout as the main bottleneck. Adding direct fine-stage mean/max access slightly worsened both clean-train and validation performance relative to v1.2. Together with v1.2's small tokenizer gain, the strongest positive Gen2 signal remains the v1.0→v1.1 resolution change.

This single-seed result does not establish statistical inferiority of multiscale readout, but it is sufficient for the preregistered practical decision: do not continue readout refinements.

### Deadline decision
For the current submission cycle, CF-HPG architecture exploration should now freeze. Best accepted Gen2 validation checkpoint remains v1.2 (`val_acc=0.5806631373641683`). With the Sep-14 paper deadline, no v1.4 optimization is justified before submission; effort should return to the already stronger LAP-GNN / Semantic ROI Graph / MGR-CNN paper program, required ablations/confirmatory evidence, RAF-DB if feasible, and manuscript closure.

No test evaluation is authorized by this review.
